torch/utils/data/datapipes/_typing.py

```diff
@@ -348,6 +348,9 @@ def _simplify_obj_name(obj) -> str:
             return str(obj.__class__.__qualname__)
         except Exception as _:
             return default_str
+    elif isinstance(obj, (list, tuple)):
+        inner = ', '.join([_simplify_obj_name(ele) for ele in obj])
+        return f"[{inner}]" if isinstance(obj, list) else f"({inner})"
```

Comment:
This can be endless data structure than can contain `DataPipe` (`dict`, `set`, all custom container objects for the custom DataPipe). Why don't we simply add a `__str__` function to `IterDataPipe` class? Then we can directly use the output from `str(obj)`.

Reply:
Yea, I think overwriting `__repr__` and `__str__` is the way to go. I checked the exception message it works.